src/ContinuousDynamicalSystem/include/BipedalLocomotion/ContinuousDynamicalSystem/LinearTimeInvariantSystem.h

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file LinearTimeInvariantSystem.h
+ * @authors Giulio Romualdi
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_CONTINUOUS_DYNAMICAL_SYSTEM_LINEAR_TIME_INVARIANT_SYSTEM_H
+#define BIPEDAL_LOCOMOTION_CONTINUOUS_DYNAMICAL_SYSTEM_LINEAR_TIME_INVARIANT_SYSTEM_H
+
+#include <memory>
+#include <vector>
+
+#include <BipedalLocomotion/ContinuousDynamicalSystem/DynamicalSystem.h>
+#include <BipedalLocomotion/ContinuousDynamicalSystem/impl/traits.h>
+
+#include <Eigen/Dense>
+
+namespace BipedalLocomotion
+{
+namespace ContinuousDynamicalSystem
+{
+
+// Forward declare for type traits specialization
+
+class LinearTimeInvariantSystem;
+}
+}
+
+// Define the internal structure of the System
+BLF_DEFINE_CONTINUOUS_DYNAMICAL_SYSTEM_INTERAL_STRUCTURE(LinearTimeInvariantSystem,
+                                                         (Eigen::VectorXd),
+                                                         (Eigen::VectorXd),
+                                                         (Eigen::VectorXd))
+
+namespace BipedalLocomotion
+{
+namespace ContinuousDynamicalSystem
+{
+
+/**
+ * LinearTimeInvariantSystem describes a MIMO linear time invariant system of the form \f$\dot{x} =
+ * Ax + Bu\f$ where \a x is the state and \a u the control input. The state, its derivative and the
+ * control input are described by vectors
+ * The LinearTimeInvariantSystem inherits from a generic DynamicalSystem where:
+ * - DynamicalSystem::State is described by an std::tuple containing:
+ *   - Eigen::VectorXd: a generic state.
+ * - DynamicalSystem::StateDerivative is described by an std::tuple containing:
+ *   - Eigen::VectorXd: a generic state derivative.
+ * - DynamicalSystem::Input is described by an std::tuple containing:
+ *   - Eigen::VectorXd: a generic control input.
+ */
+class LinearTimeInvariantSystem : public DynamicalSystem<LinearTimeInvariantSystem>
+{
+    Eigen::MatrixXd m_A;
+    Eigen::MatrixXd m_B;
+
+    bool m_isInitialized{false};
+
+    State m_state;
+    Input m_controlInput;
+
+public:
+    /**
+     * Set the system matrices.
+     * @param A the A matrix.
+     * @param B the B matrix.
+     * @return true in case of success, false otherwise.
+     */
+    bool setSystemMatrices(const Eigen::Ref<const Eigen::MatrixXd>& A,
+                           const Eigen::Ref<const Eigen::MatrixXd>& B);
+
+    /**
+     * Initialize the Dynamical system.
+     * @param handler pointer to the parameter handler.
+     * @return true in case of success/false otherwise.
+     */
+    bool initalize(std::weak_ptr<ParametersHandler::IParametersHandler> handler);
```

Comment:
Typo: ``initalize``

Reply:
Oh noooooooooooooo :cry:

Reply:
Fixed in 84e69ad (The same for the others)